libsolidity/ast/Types.cpp

```diff
@@ -347,6 +347,11 @@ TypePointer IntegerType::binaryOperatorResult(Token::Value _operator, TypePointe
 	// All integer types can be compared
 	if (Token::isCompareOp(_operator))
 		return commonType;
+	// Disable >>> here.
+	if (_operator == Token::SHR)
+		return TypePointer();
+	if (Token::isShiftOp(_operator) && !isAddress()) // && !_other->isAddress())
```

Comment:
I think you have to do this whole thing even before line 342. The following should be possible (please add a test):
```
uint x = 70;
int8 y = 2;
x << y;
```
`x` and `y` do not have a common type, and so currently, it would fail in line 345.

Yes, we should perform some more checks for the other side. I would say rational number is fine as long as it is an integer. Fixed point is fine, but we have to perform an integer check at run-time. Integer is fine as long as it is not an address.